wpimath/src/main/java/edu/wpi/first/math/Matrix.java

```diff
@@ -738,4 +741,18 @@ public boolean equals(Object other) {
   public int hashCode() {
     return Objects.hash(m_storage);
   }
+
+  /**
+   * Creates an implementation of the {@link Protobuf} interface for matrices.
+   *
+   * @param <R> The number of rows of the matrices this serializer processes.
+   * @param <C> The number of cols of the matrices this serializer processes.
+   * @param rows The number of rows of the matrices this serializer processes.
+   * @param cols The number of cols of the matrices this serializer processes.
+   * @return The protobuf implementation.
+   */
+  public static <R extends Num, C extends Num> MatrixProto<R, C> getProto(
+      Nat<R> rows, Nat<C> cols) {
+    return new MatrixProto<>(rows, cols);
+  }
```

Comment:
It should be possible to do struct serializations of Matrix and Vector, it would just be necessary to have the struct name depend on the generic/template parameter values so it's unique (e.g. `Matrix_R_C`, `Vector<size>`)?

Reply:
Yeah, that sounds like it should work! I'd probably do underscore separation for everything (both `Matrix_R_C` and `Vector_R`) to be consistent. (Also because `<` isn't (currently) allowed to be in the type identifier. The syntax of an identifier name isn't actually described in [the spec](https://github.com/wpilibsuite/allwpilib/blob/cc02a948a0f1a23a758e7437c07bb7682a724cce/wpiutil/doc/struct.adoc), but it's implied to be at least similar to C identifier syntax (which doesn't allow `<`) and [SchemaParser](https://github.com/wpilibsuite/allwpilib/blob/cc02a948a0f1a23a758e7437c07bb7682a724cce/wpiutil/src/main/native/cpp/struct/SchemaParser.cpp#L97) sticks to that convention. It'd be interesting to add that to the schema, but realistically the better approach would just be to describe how to do the name mangling for templates, which we could just say would be to use `_` between the template name and the arguments, as well as between each argument. (Or maybe `__` before the template arguments if we want to be fancy))

Reply:
Yeah, that sounds like it should work! I'd probably do underscore separation for everything (both `Matrix_R_C` and `Vector_R`) to be consistent. (Also because `<` isn't (currently) allowed to be in the type identifier. The syntax of an identifier name isn't actually described in [the spec](https://github.com/wpilibsuite/allwpilib/blob/cc02a948a0f1a23a758e7437c07bb7682a724cce/wpiutil/doc/struct.adoc), but it's implied to be at least similar to C identifier syntax (which doesn't allow `<`) and [SchemaParser](https://github.com/wpilibsuite/allwpilib/blob/cc02a948a0f1a23a758e7437c07bb7682a724cce/wpiutil/src/main/native/cpp/struct/SchemaParser.cpp#L97) sticks to that convention. It'd be interesting to add that to the schema, but realistically the better approach would just be to describe how to do the name mangling for templates, which we could just say would be to use `_` between the template name and the arguments, as well as between each argument. (Or maybe `__` before the template arguments if we want to be fancy))